plugins/www/Optimove.js

```diff
@@ -77,139 +99,199 @@ function nativeMessageHandler(message) {
-    });
-  },
-  inAppPresentInboxMessage: function (inAppInboxItem) {
-    return new Promise((resolve, reject) => {
-      exec(resolve, reject, "OptimoveSDKPlugin", "inAppPresentInboxMessage", [
-        inAppInboxItem.id,
-      ]);
-    });
-  },
-  inAppDeleteMessageFromInbox: function (inAppInboxItem) {
-    return new Promise((resolve, reject) => {
-      exec(
-        resolve,
-        reject,
-        "OptimoveSDKPlugin",
-        "inAppDeleteMessageFromInbox",
-        [inAppInboxItem.id]
-      );
-    });
-  },
-
-  setOnInboxUpdatedHandler: function (handler) {
-    currentConfig["inAppInboxUpdatedHandler"] = handler;
-  },
-
-  setPushOpenedHandler(pushOpenedHandler) {
-    currentConfig["pushOpenedHandler"] = pushOpenedHandler;
-    if (pushOpenedHandler != null) {
-      checkIfPendingPushExists();
-    }
-  },
+    setUserId: function (userId) {
+        return new Promise((resolve, reject) => {
+            if (!isNonEmptyString(userId)) {
+                reject("Invalid user id");
+                return;
+            }
+            exec(resolve, reject, "OptimoveSDKPlugin", "setUserId", [userId]);
+        });
+    },
 
-  setPushReceivedHandler(pushReceivedHandler) {
-    currentConfig["pushReceivedHandler"] = pushReceivedHandler;
-  },
+    setUserEmail: function (userEmail) {
+        return new Promise((resolve, reject) => {
+            if (!isNonEmptyString(userEmail)) {
+                reject("Invalid user email");
+                return;
+            }
+            exec(resolve, reject, "OptimoveSDKPlugin", "setUserEmail", [userEmail]);
+        });
+    },
 
-  setInAppDeepLinkHandler(inAppDeepLinkHandler) {
-    currentConfig["inAppDeepLinkHandler"] = inAppDeepLinkHandler;
-  },
-};
-module.exports = Optimove;
+    /* eventParams is nullable*/
+    reportEvent: function (eventName, eventParams) {
+        return new Promise((resolve, reject) => {
+            if (!isNonEmptyString(eventName)) {
+                reject("Invalid event name");
+                return;
+            }
+            exec(resolve, reject, "OptimoveSDKPlugin", "reportEvent", [
+                eventName,
+                eventParams,
+            ]);
+        });
+    },
+
+    /* screenCategory parameter is nullable*/
+    reportScreenVisit: function (screenName, screenCategory) {
+        return new Promise((resolve, reject) => {
+            if (!isNonEmptyString(screenName)) {
+                reject("Invalid screen name");
+                return;
+            }
+            exec(resolve, reject, "OptimoveSDKPlugin", "reportScreenVisit", [
+                screenName,
+                screenCategory,
+            ]);
+        });
+    },
+
+    registerUser: function (userId, userEmail) {
+        return new Promise((resolve, reject) => {
+            var invalidRegister = false;
+            if (!isNonEmptyString(userId)) {
+                reject("Invalid user id");
+                invalidRegister = true;
```

Comment:
better to return here? Once it's rejected it stays rejected, and (maybe) message doesnt change